From: Michael Olbrich <m.olbrich@pengutronix.de>
Date: Fri, 1 Nov 2019 16:58:55 +0100
Subject: [PATCH] fix building on ARM

The build-system currently assumes that 'arm' is ARMv8 as well. This is
completely broken. Fix it by ensuring, that ARMv8 code is only build for
aarch64.

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
 nss/lib/freebl/Makefile | 16 ----------------
 1 file changed, 16 deletions(-)

diff --git a/nss/lib/freebl/Makefile b/nss/lib/freebl/Makefile
index 5943fb377894..53bf55b57fc8 100644
--- a/nss/lib/freebl/Makefile
+++ b/nss/lib/freebl/Makefile
@@ -123,22 +123,6 @@ ifeq ($(CPU_ARCH),aarch64)
     DEFINES += -DUSE_HW_AES
     EXTRA_SRCS += aes-armv8.c gcm-aarch64.c
 endif
-ifeq ($(CPU_ARCH),arm)
-    ifdef CC_IS_CLANG
-        DEFINES += -DUSE_HW_AES
-        EXTRA_SRCS += aes-armv8.c
-    else ifeq (1,$(CC_IS_GCC))
-        # Old compiler doesn't support ARM AES.
-        ifneq (,$(filter 4.9,$(word 1,$(GCC_VERSION)).$(word 2,$(GCC_VERSION))))
-            DEFINES += -DUSE_HW_AES
-            EXTRA_SRCS += aes-armv8.c
-        endif
-        ifeq (,$(filter 0 1 2 3 4,$(word 1,$(GCC_VERSION))))
-            DEFINES += -DUSE_HW_AES
-            EXTRA_SRCS += aes-armv8.c
-        endif
-    endif
-endif
 
 ifeq ($(OS_TARGET),OSF1)
     DEFINES += -DMP_ASSEMBLY_MULTIPLY -DMP_NO_MP_WORD
